### PR TITLE
Send ECS task lifecycle events

### PR DESCRIPTION
**Pull request by kangyili**
### What does this PR do?

Based on https://github.com/DataDog/datadog-agent/pull/23250, this PR updates lifecycle event check to send ecs task events

https://github.com/DataDog/agent-payload/pull/282

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: releasenotes/notes/send-ecs-task-lifecycle-events-b09a2c367c864c71.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    Send ECS task lifecycle events in container lifecycle check
```
Comment (alai97) [inline releasenotes/notes/send-ecs-task-lifecycle-events-b09a2c367c864c71.yaml:11]:
```suggestion
    Send ECS task lifecycle events in the container lifecycle check.
```

**Review by alai97 — APPROVED** [review]
Looks good, one suggestion for the release note.

**Review by hush-hush — APPROVED** [review]
Looks good for the files owned by ASC

File: pkg/collector/corechecks/containerlifecycle/check.go [hunk 2]
```diff
@@ -155,3 +178,30 @@ func Factory(store workloadmeta.Component) optional.Option[func() check.Check] {
...
 }
+
+// sendFargateTaskEvent sends Fargate task lifecycle event at the end of the check
+func (c *Check) sendFargateTaskEvent() {
+	shouldSend := ddConfig.Datadog.GetBool("container_lifecycle.ecs_task_event.enabled") &&
+		ddConfig.IsECSFargate()
+
+	if !shouldSend {
+		return
+	}
```
Comment (L3n41c) [inline pkg/collector/corechecks/containerlifecycle/check.go:189]:
```suggestion
	if !ddConfig.Datadog.GetBool("container_lifecycle.ecs_task_event.enabled") ||
		!ddConfig.IsECSFargate() {
		return
	}
```

**Review by L3n41c — APPROVED** [review]

File: pkg/collector/corechecks/containerlifecycle/check.go (hunk 2 above)
Comment (kangyili) [reply pkg/collector/corechecks/containerlifecycle/check.go]: https://github.com/DataDog/datadog-agent/pull/21984/commits/3b5863b9ff336e362f19f45910a8154f2e09d637

**Pull request merged**